client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs

```diff
@@ -576,6 +584,37 @@ public async Promise UpdateServiceReferences(string serviceName, List<AssemblyDe
 			UsamLogger.Log($"Finished updating references");
 		}
 
+		public async Promise UpdateServiceStoragesDependencies(IBeamoServiceDefinition service, List<string> dependencies)
+		{
+			var serviceName = service.BeamoId;
+			var currentDependencies = service.Dependencies;
+
+			var dependenciesToRemove = currentDependencies.Where(dep => !dependencies.Contains(dep)).ToList();
+			var dependenciesToAdd = dependencies.Where(dep => !currentDependencies.Contains(dep)).ToList();
+
+			//TODO: can this be made asynchronous? not sure since all are changing the same csproj file
+
+			foreach (string dep in dependenciesToRemove)

```

Comment:
Don't like this `await` inside foreach approach. But I'm afraid to introduce problems if I start asynchronous tasks that are changing the same file, without the possibility to lock on the file itself from Unity